### DISCUSSION TITLE
"equal or after" scenario in "after" and "before" validator

### DISCUSSION THREAD

**Discussion by Vong3432** [adonisjs/core#3451] [Ideas]
Hello there 👋,  

I have been using the validators quite a while and so far I think it is very easy and good to use. Recently I have faced some scenarios where a date have to be either same with today or after today.

### Given
- Current date: **2021-12-29**
- John wants to have a holiday on **2021-12-29**
- Today is **2021-12-29** 

### Validator requirement
- The date should be either today or after today. But any date before today is invalid.

#### My current approach 
```
// valid 
rules.after(-1, "day")

// invalid
rules.after(0, "day")
rules.after("today") 
```

#### Ideal approach
```
rules.after("today", { equalTo: true })
```

#### Documentation 
<img width="450" alt="image" src="https://user-images.githubusercontent.com/36991424/147662565-1ba757fc-ce7f-4e5c-b2ce-b58062a18375.png">

Although my current approach worked and the documentation did mention how we can add a custom ref to handle more advanced usage, but the code itself looks kinda weird and not easy to understand at first sight. 

Would it be better if we can have an extra optional parameters in the `after` validator to handle these scenarios?

**Comment by thetutlage** [comment]
How about adding two new rules. `beforeOrEqual` and `afterOrEqual`? In your case it will be `afterOrEqual('today')`

**Reply by Vong3432** [reply]
Looks good!

**Reply by thetutlage** [reply]
Do you think you can create a PR for the same?

**Reply by Vong3432** [reply]
I can give it a try. Should that PR be created in [adonis/validator](https://github.com/adonisjs/validator) repo?
